File: C++/Hard/WrongAnswer/minimum-time-to-remove-all-cars-containing-illegal-goods_3.cpp
class Solution {
public:
    int minimumTime(string s) {
        
        int n = s.size();
        int i = 0;
        int j = n - 1;
        while (i < n && s[i] == '0') i++;
        while (j >= 0 && s[j] == '0') j--;
        if (i >= j) return 0;
        int ans = 2;
        for (int k = i; k <= j; k++) {
            if (s[k] == '1') {
                ans += 2;
                while (k <= j && s[k] == '1') k++;
            }
        }
        return ans;
    }
};

File: Rust/Hard/WrongAnswer/minimum-time-to-remove-all-cars-containing-illegal-goods_3.rs
impl Solution {
    pub fn minimum_time(s: String) -> i32 {
        
        let mut res = 0;
        let mut count = 0;
        for c in s.chars() {
            if c == '0' {
                count = 0;
            } else {
                count += 1;
                if count % 2 == 1 {
                    res += 2;
                }
            }
        }
        res - if count % 2 == 1 { 1 } else { 0 }
    }
}